### ISSUE TITLE
index out of bounds in Tabs

### ISSUE THREAD

**Issue by StillGreen-san** [iced-rs/iced_aw#213]
if Tabs is used as the root element the program will panic with an out of bounds index (on an empty Tree).
unless a different element has been displayed before (like a loading screen).

```
thread 'main' panicked at [..]\iced_aw\src\native\tabs.rs:315:22:
index out of bounds: the len is 0 but the index is 1
```
_thats in diff() in impl Widget for Tabs, but there are more functions effected by this_

if Tabs is put in a column for example this does not happen. wrapping it in a container still crashes tho.

i dont know how to best address this, i tried to handle the empty case in all the functions that use it, but that resulted in the tab contents not showing.

**Comment by genusistimelord** [comment]
can you give the tabs_fix branch a try and see if it still errors?

**Comment by genusistimelord** [comment]
ok then i fixed those. my compiler is not working ATM so it wasn't displaying the errors.

**Comment by StillGreen-san** [comment]
there is still an `!` in that if that should not be there, but without that it works (did some more tests with&without wrapping elements & loading screens, all work)
```diff
Index: src/native/tabs.rs
@@ -314,3 +314,3 @@
     fn diff(&self, tree: &mut Tree) {
-        if !tree.children.is_empty() {
+        if tree.children.is_empty() {
             let tabs = Tree {
```

_cannot speak for the modal changes, that have made there way into that branch, tho_

but im wondering, is there a possibility for `tree` to only have 1 element? if so a checking `len() != 2` might be better?
or could `tree` have 2 or more elements that are however not the tab children?

**Comment by genusistimelord** [comment]
this is now fixed in the last PR #218
